assets/js/script.js

```diff
@@ -0,0 +1,304 @@
+const startButton = document.querySelector("#start-button");
+const startContainer = document.querySelector("#start-container");
+const mainContainer = document.querySelector("#main-container");
+
+// render question container
+const renderQuestion = function () {
+  // get question
+  const question = questionsArray[questionIndex];
+
+  // target questionContainer for removal and click events
+  const questionContainer = document.createElement("div");
+  questionContainer.setAttribute("id", "question-container");
+  questionContainer.setAttribute("data-answer", question.correctAnswer);
+
+  //   create buttons div
+  const buttonContainer = document.createElement("div");
+  buttonContainer.setAttribute("id", "answer-buttons");
+
+  //   create answer buttons and append with for each
+  const renderAnswers = function (eachAnswer, index) {
+    const answerButton = document.createElement("button");
+    answerButton.setAttribute("class", "answerButton");
+    answerButton.setAttribute("id", index);
+    answerButton.setAttribute("data-option", eachAnswer);
+    answerButton.textContent = eachAnswer;
+    buttonContainer.appendChild(answerButton);
+  };
+
+  question.answers.forEach(renderAnswers);
+  currentQuestion = questionsArray[questionIndex];
+
+  //   create question text
+  const questionText = document.createElement("h2");
+  questionText.setAttribute("id", "question-text");
+  questionText.textContent = currentQuestion.question;
+
+  //  append 2 children to questionContainer
+  questionContainer.appendChild(questionText);
+  questionContainer.appendChild(buttonContainer);
+
+  // verify answer click event
+  questionContainer.addEventListener("click", verifyAnswer);
+
+  mainContainer.appendChild(questionContainer);
+};
+
+const verifyAnswer = function (event) {
+  const currentTarget = event.currentTarget;
+  const target = event.target;
+
+  const userClicked = target.getAttribute("data-option");
+  const correctAnswer = currentTarget.getAttribute("data-answer");
+
+  if (questionIndex === questionsArray.length - 1) {
+    if (userClicked === correctAnswer) {
+      isGameCompleted = true;
+      renderScoreForm();
+    }
+    return;
+  }
+
+  //   check if click is on the correct answer button
+  if (userClicked === correctAnswer) {
+    //   if true then increment questionIndex value
+    if (questionIndex <= questionsArray.length - 1) {
+      removeQuestion();
+      questionIndex += 1;
+      renderQuestion();
+    }
+    // deduct time
+  } else {
+    counter -= 5;
+  }
+};
+
+// remove question container
+const removeQuestion = function () {
+  document.getElementById("question-container").remove();
+};
+
+// renderGameOver
+const renderGameOver = function () {
+  counterSpan.textContent = 0;
+  removeQuestion();
+
+  // render link to start
+  const newStartLink = document.createElement("a");
+  newStartLink.setAttribute("href", "./index.html");
+  newStartLink.textContent = "Start Quiz";
+
```

Comment:
I think it's this button which says Start Quiz but it takes you to the home page. Consider changing the name of this or amending the functionality.